haskell/test/Test/Ligo/BaseDAO/Proposal.hs

```diff
@@ -42,156 +42,167 @@ test_BaseDAO_Proposal =
-      , nettestScenarioOnEmulator "can flush 2 proposals that got accepted" $
-          \_emulated ->
-            uncapsNettest $ flushAcceptedProposalsWithAnAmount (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      , nettestScenarioOnEmulator "can flush proposals that got rejected due to not meeting quorum_threshold" $
-          \_emulated ->
-            uncapsNettest $ flushRejectProposalQuorum (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      , nettestScenarioOnEmulator "can flush proposals that got rejected due to negative votes" $
-          \_emulated ->
-            uncapsNettest $ flushRejectProposalNegativeVotes (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      , nettestScenarioOnEmulator "flush should not affecting ongoing proposals" $
-          \_emulated ->
-            uncapsNettest $ flushNotAffectOngoingProposals
-            (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      , nettestScenarioOnEmulator "flush with bad cRejectedProposalReturnValue" $
-          \_emulated ->
-            uncapsNettest $ flushWithBadConfig (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      -- TODO [#15]: admin burn proposer token and test "flush"
+
+      , nettestScenarioOnEmulatorCaps "can flush 2 proposals that got accepted" $
+          flushAcceptedProposalsWithAnAmount (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "can flush proposals that got rejected due to not meeting quorum_threshold" $
+          flushRejectProposalQuorum (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "can flush proposals that got rejected due to negative votes" $
+          flushRejectProposalNegativeVotes (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "flush should not affect proposals that cannot be flushed yet" $
+          flushProposalFlushTimeNotReach (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "flush should skip expired proposals" $
+          flushSkipExpiredProposal (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "flush with bad cRejectedProposalReturnValue" $
+          flushWithBadConfig (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      -- TODO [#15]: dodAdmin burn proposer token and test "flush"
 
       -- TODO [#38]: Improve this when contract size is smaller
-      , nettestScenarioOnEmulator "flush and run decision lambda" $
-          \_emulated ->
-            uncapsNettest $ flushDecisionLambda (originateLigoDaoWithConfigDesc dynRecUnsafe)
-      , nettestScenarioOnEmulator "can drop proposals" $
-          \_emulated ->
-            uncapsNettest $ dropProposal (originateLigoDaoWithConfigDesc dynRecUnsafe)
+      , nettestScenarioOnEmulatorCaps "flush and run decision lambda" $
+          flushDecisionLambda (originateLigoDaoWithConfigDesc dynRecUnsafe)
+
+      , nettestScenarioOnEmulatorCaps "can drop proposals" $
```

Comment:
```suggestion
      , nettestScenarioOnEmulatorCaps "can drop proposals, only when allowed" $
```
I think this should be updated as I see you have extended the `dropProposal` test to include the check for forbidden cases as well.